lib/client-assets.php

```diff
@@ -1216,14 +1216,24 @@ function polyfill_blocks_module_in_scripts() {
  */
 function editor_color_palette_slugs( $color_palette ) {
 	$new_color_palette = array();
+	$is_doing_it_wrong = false;
 
 	foreach ( $color_palette as $color ) {
 		if ( ! isset( $color['slug'] ) ) {
 			$color['slug'] = esc_js( $color['name'] );
+			$is_doing_it_wrong = true;
 		}
 
 		$new_color_palette[] = $color;
 	}
 
+	if( $is_doing_it_wrong ) {
+		_doing_it_wrong(
+			'add_theme_support/editor-color-palette',
```

Comment:
This needs to be the function name only, i.e. `add_theme_support()`. I agree though that it makes sense to highlight the theme feature, so maybe in the description you could say "Each color specified for the editor color palette...".